Davtyan dev

Davt2000 opened the pull request:
Реквест к РК3

esuwu committed:
First commit

esuwu committed:
Готово к проверке

esuwu committed:
Готово к проверке

Davt2000 committed:
Created the base of Archiver and TypeIdentifier for team review. Most likely next changes will be made soon.

Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Davt2000 committed:
Created the base of Archiver and TypeIdentifier for team review. Most likely next changes will be made soon.

Signed-off-by: Davt2000 <davtart2000@yandex.ru>

esuwu committed:
Rev2

esuwu committed:
Review2

Davt2000 committed:
Merge remote-tracking branch 'origin/Davtyan_dev' into Davtyan_dev

Davt2000 committed:
Created the base of Archiver and TypeIdentifier for team review. Most likely next changes will be made soon.

Signed-off-by: Davt2000 <davtart2000@yandex.ru>

esuwu committed:
Лови

esuwu committed:
Лови

Davt2000 committed:
Merge remote-tracking branch 'origin/InputOutput' into Davtyan_dev

Davt2000 committed:
Added a cmake extension to link boost and configured it. Added gitignore.
Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Davt2000 committed:
Cleared my branch from not my code.
Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Davt2000 committed:
Cleared my branch from not my code.
Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Davt2000 committed:
Merge remote-tracking branch 'origin/Davtyan_dev' into Davtyan_dev

Davt2000 committed:
removed IDE files

Davt2000 committed:
Fully implemented TypeIdentifier. Ready to use.
Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Davt2000 committed:
Remade base of Archiver
Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Davt2000 committed:
Cleaned up the TypeIdentifier and fully implemented packing the archive.
Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Signed-off-by: Davt2000 <davtart2000@yandex.ru>

leshiy1295 commented:
плохо, что в истории коммитов не только Ваши изменения :(

leshiy1295 commented on ArchiveFile.cpp:
```diff
@@ -0,0 +1,83 @@
+}
+
+void ArchiveFile::WriteSignature () {
+  Write (0x74); // TPRK archive file signature (tprk)
```
обсудили на РК, что это не универсально, и всё равно возможны коллизии => стоит перейти на схему <datasize><data>

leshiy1295 commented on ArchiveFile.cpp:
```diff
@@ -0,0 +1,83 @@
+ArchiveFile::ArchiveFile (std::string path) :Output (path), filepath (std::move(path)){
+  fout.open(filepath, std::ios::binary);
+  if (!fout.is_open()) {
+    std::cout << "can't open or create file" << filepath << std::endl;
```
не нужно использовать cout/cerr просто так в проекте
используйте отдельную управляемую и централизованную логгирующую сущность

leshiy1295 commented on ArchiveFile.hpp:
```diff
@@ -0,0 +1,45 @@
+#ifndef _ARCHIVEFILE_HPP_
+#define _ARCHIVEFILE_HPP_
+
+#include <vector>
```
выносить в header особенности внутренней реализации - плохо
используйте обёртки над стандартными контейнерами + PIMPL

leshiy1295 commented on Archiver.cpp:
```diff
@@ -0,0 +1,38 @@
+
+std::map<std::string, std::string> Archiver::Unpack (const std::string& path_to_archive)
+{
+  return std::map<std::string, std::string> ();
```
не реализовано...

leshiy1295 commented on Archiver.cpp:
```diff
@@ -0,0 +1,38 @@
+
+std::string Archiver::Pack (const std::map<std::string, std::string>& compressed_data)
+{
+  ArchiveFile archive("Archive.tprk");
```
название захардкожено

leshiy1295 commented on Archiver.cpp:
```diff
@@ -0,0 +1,38 @@
+
+
+
+  return std::__cxx11::string ();
```
""

leshiy1295 commented on Archiver.hpp:
```diff
@@ -0,0 +1,20 @@
+
+class Archiver{
+ public:
+  static std::map<std::string, std::string> Read(const std::string& path_to_archive);
```
отдавать наружу map - совсем плохо :(

leshiy1295 commented on Input.cpp:
```diff
@@ -0,0 +1,38 @@
+        value = (unsigned char)buff;
+        count++;
+        return true;
+    } else {
```
else после return не нужен

leshiy1295 commented on Input.h:
```diff
@@ -0,0 +1,31 @@
+#include "IInputStream.h"
+#include "IGetFileSize.h"
+
+typedef unsigned char byte;
```
в стандарте есть std::byte
+ этот код я уже видел у Вашего напарника

leshiy1295 commented on TypeIdentifier.cpp:
```diff
@@ -0,0 +1,29 @@
+  auto handle = magic_open(MAGIC_MIME_TYPE);
+  magic_load(handle, MIME_DB);
+
+  std::string type = magic_file (handle, path.c_str());
```
названия методов должны быть информативными

Davt2000 replied:
Этот метод из встроенной библитотеки magic

leshiy1295 commented on TypeIdentifier.cpp:
```diff
@@ -0,0 +1,29 @@
+
+  std::string type = magic_file (handle, path.c_str());
+  std::vector<std::string> out;
+  boost::split(out, type, [](char c){return c == '/';});
```
использовать boost внутри библиотеки запрещено по правилам
для split можно использовать регулярные выражения
для разбиения path - использовать std::filesystem

leshiy1295 commented on TypeIdentifier.cpp:
```diff
@@ -0,0 +1,29 @@
+std::vector<std::string> TypeIdentifier::SignatureDetect(const std::vector<std::string>& list){
+  std::vector<std::string> out;
+
+  for(const auto & it : list){
```
используйте функциональный стиль - transform/back_inserter и пр.

leshiy1295 commented on TypeIdentifier.hpp:
```diff
@@ -0,0 +1,30 @@
+#include <map>
+#include <string>
+#include <boost/algorithm/string.hpp>
+#include <magic.h>
```
использовать внешние зависимости внутри Вашей библиотеки нельзя

leshiy1295 reviewed (changes requested):
нужно избавиться от внешних зависимостей и добавить интерфейсные развязки
когда исправите все замечания - можете отправить код на повторное ревью

Davt2000 committed:
Purged boost from TypeIdentifier implementation. Fixed numbers writing and made entry system clear
Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Davt2000 committed:
Now really purged boost
Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Davt2000 committed:
Optimized archive structure to make it more machine-readable
Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Davt2000 committed:
Made a way to read a data using ArchiveFile
Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Davt2000 committed:
Fixed TypeIdentifier for compatibility.
Implemented reading(not unpacking) the archive. NOT TESTED
Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Davt2000 committed:
Minor improves
Added content start pointer to archive
Started to change general architecture of Archiver
Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Davt2000 committed:
Review fix

Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Davt2000 committed:
Core changes
Fixed archive reading and made core changes to architecture(ArchiveFile split into ArchiveFileReader and ArchiveFileWriter).
Signed-off-by: Davt2000 <davtart2000@yandex.ru).

Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Davt2000 committed:
Dearchivator added and tested
Signed-off-by: Davt2000 <davtart2000@yandex.ru).

Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Davt2000 committed:
Added invalid archive exception
Signed-off-by: Davt2000 <davtart2000@yandex.ru).

Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Davt2000 committed:
Update ArchiveFileWriter.cpp

Foreshadowing a fix of path writing into the archive

Davt2000 committed:
Added binary files cleanup with single file and vector of files implementation. NOT TESTED
Signed-off-by: Davt2000 <davtart2000@yandex.ru).

Signed-off-by: Davt2000 <davtart2000@yandex.ru>

Davt2000 committed:
Fix an archive corruption. Attempt 1
Signed-off-by: Davt2000 <davtart2000@yandex.ru).

Signed-off-by: Davt2000 <davtart2000@yandex.ru>

leshiy1295 commented on ArchiveFileReader.cpp:
```diff
@@ -0,0 +1,72 @@
+      Read(stub);
+      out.push_back(stub);
+    };
+  if(out[4] != '\016' || out.substr(0, 3) == "tprk")
```
много магических констант :(

leshiy1295 commented on buff.cpp:
```diff
@@ -0,0 +1,19 @@
+#include <iomanip>
+
+// 8 bytes. AAAA AAAA AAAA AAAA.
+// 1111000011110000111100001111000011110000111100001111000011110000
```
сложно
